avocado/utils/pci.py

```diff
@@ -59,6 +59,22 @@ def get_pci_addresses():
         return addresses
 
 
+def get_pci_hbas():
+    """
+    Gets list of PCI hbas, first pci address in a hba, in the system.
+    Does not return the PCI Bridges/Switches.
+    e.g: for pci address 0000:00:14.0 and 0000:00:14.1, we return 0000:00:14.0
+
+    :return: list of full PCI hbas including domain (0000:00:14.0)
+    """
```

Comment:
Hello @narasimhan-v, I'm wondering what functionality are you looking for. HBA means host bus adapter, which is mainly associated either with disk or nic cards. It can also mean ports, but it should not report other devices like random number generators and other devices not providing a bus. Therefor either the name, or the functionality should be a bit different.

Now how your code should behave when the device does not use `.0` address? Currently it reports all of the devices (eg: `[1.1, 1.2] => [1.1, 1.2]`). Is this expected, or do you always want to report the first device (`[1.1]`)?

Also are you sure `lspci` is always sorted? If so then it works well, otherwise it might produce incorrect results (`[1.2, 1.0] => [1.2, 1.2]` while `[1.0, 1.2] => [1.0]`). Anyway I think we can relate to `lspci` being sorted for now and fix it if we found incorrectly sorted output...